airflow/executors/kubernetes_executor.py

```diff
@@ -364,7 +364,7 @@ def process_status(self, pod_id, status, labels, resource_version):
             self.log.info('Event: %s Pending', pod_id)
         elif status == 'Failed':
             self.log.info('Event: %s Failed', pod_id)
-            self.watcher_queue.put((pod_id, State.FAILED, labels, resource_version))
```

Comment:
I think this is too simplistic and will miss cases where the pod hard fails and hasn't updated the state (i.e. if it gets OOM killed) - likely this would end up leaving tasks in the running state.

Reply:
Thanks @ashb but in such scenarios wouldn't scheduler mark them as zombie and would appropriately update their state like either as failed or up_for_retry etc.
Currently watcher is updating task state and Scheduler is ignoring them.

Reply:
@dimberman Can you Please have a look and see if it makes sense for K8 Executor.